internal/app/synchronizer/evaluator_client.go

```diff
@@ -119,10 +119,15 @@ func (ec *grcpEvaluatorClient) evaluate(ctx context.Context, pc <-chan []*pb.Mat
 	}
 
 	results := []string{}
+	matchIDs := map[string]bool{}
 
 	eg.Go(func() error {
 		for proposals := range pc {
 			for _, proposal := range proposals {
+				if matchIDs[proposal.GetMatchId()] {
```

Comment:
`_, ok := matchIDs[proposal.GetMatchId()]; ok`

It's possible that it could have already been returned by the evaluator